types/rdf-js/index.d.ts

```diff
@@ -174,27 +174,27 @@ export type Quad_Graph = DefaultGraph | NamedNode | BlankNode | Variable;
 /**
  * An RDF quad, taking any Term in its positions, containing the subject, predicate, object and graph terms.
  */
-export interface BaseQuad {
+export interface BaseQuad<S extends Term = Term, P extends Term = Term, O extends Term = Term, G extends Term = Term> {
```

Comment:
I don't think you have to genarlise S/P/O to get the "real triple". 

How about something like this instead:

```ts
export interface BaseQuad<TGraph extends Term = Term> {
  // ...

  graph: TGraph;
}

export interface Quad extends BaseQuad<Quad_Graph> {
  // ...
}

export interface Triple extends Quad<DefaultGraph> {
}
```